guide/coding_style_guide.md

```diff
@@ -0,0 +1,32 @@
+# Coding style
+
+- Use Elixir formatter (`mix format`). `.formatter.exs` should contain `:membrane_core` in `:import_deps`.
+- Use Dialyzer (via `:dialyxir` dependency and `mix dialyzer` command) and make sure it shows no warnings.
+- Indent with 2 spaces, use `LF` as end of line and `utf-8` encoding
+- For native code, use `clang-format` with config (`.clang-format`):
+
+  ```
+  BasedOnStyle: LLVM
+  IndentWidth: 2
+  ```
+
+- Avoid ambiguous abbreviations (e.g. `res`) and unnamed ignored variables (e.g. `def handle_init(_)` -> `def handle_init(_options)`)
+- When implementing something defined in some kind of document or standard (e.g. RFC) try to match variable names as closely as possible to the standard and always provide a reference in a comment
+- Every public module must have `@moduledoc` and its public functions both `@doc` and `@spec`
+- Always use `@impl` when implementing behaviour. Apply it to every function clause to avoid situations where the last clause has a typo that is not reported as warning since its treated as different function
```

Comment:
- mention sorting uses, imports and aliases
- link https://hexdocs.pm/elixir/writing-documentation.html
- link https://hexdocs.pm/elixir/master/library-guidelines.html

Reply:
and mention grouping and nesting modules in docs